Index: vcglib/vcg/complex/algorithms/inertia.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/inertia.h
+++ vcglib/vcg/complex/algorithms/inertia.h
@@ -24,8 +24,8 @@
 #define _VCG_INERTIA_
 
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <Eigen/Core>
+#include <Eigen/Eigenvalues>
 #include <vcg/complex/algorithms/update/normal.h>
 
 namespace vcg
Index: vcglib/vcg/complex/algorithms/parametrization/poisson_solver.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/parametrization/poisson_solver.h
+++ vcglib/vcg/complex/algorithms/parametrization/poisson_solver.h
@@ -24,7 +24,7 @@
 #ifndef VCG_POISSON_SOLVER
 #define VCG_POISSON_SOLVER
 
-#include <eigenlib/Eigen/Sparse>
+#include <Eigen/Sparse>
 
 #include <vcg/complex/algorithms/clean.h>
 #include <vcg/complex/algorithms/update/bounding.h>
Index: vcglib/vcg/complex/algorithms/update/curvature.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/update/curvature.h
+++ vcglib/vcg/complex/algorithms/update/curvature.h
@@ -32,7 +32,7 @@
 #include <vcg/complex/algorithms/point_sampling.h>
 #include <vcg/complex/algorithms/intersection.h>
 #include <vcg/complex/algorithms/inertia.h>
-#include <eigenlib/Eigen/Core>
+#include <Eigen/Core>
 
 namespace vcg {
 namespace tri {
Index: vcglib/vcg/complex/algorithms/update/curvature_fitting.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/update/curvature_fitting.h
+++ vcglib/vcg/complex/algorithms/update/curvature_fitting.h
@@ -38,11 +38,11 @@
 #include <vcg/complex/algorithms/inertia.h>
 #include <vcg/complex/algorithms/nring.h>
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/QR>
-#include <eigenlib/Eigen/LU>
-#include <eigenlib/Eigen/SVD>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <Eigen/Core>
+#include <Eigen/QR>
+#include <Eigen/LU>
+#include <Eigen/SVD>
+#include <Eigen/Eigenvalues>
 
 
 namespace vcg {
Index: vcglib/vcg/complex/algorithms/update/fitmaps.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/update/fitmaps.h
+++ vcglib/vcg/complex/algorithms/update/fitmaps.h
@@ -34,10 +34,10 @@
 #include <vcg/complex/algorithms/update/bounding.h>
 #include "vcg/complex/algorithms/update/curvature_fitting.h"
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/QR>
-#include <eigenlib/Eigen/LU>
-#include <eigenlib/Eigen/SVD>
+#include <Eigen/Core>
+#include <Eigen/QR>
+#include <Eigen/LU>
+#include <Eigen/SVD>
 
 #include <vcg/complex/algorithms/nring.h>
 
Index: vcglib/vcg/math/eigen.h
===================================================================
--- vcglib.orig/vcg/math/eigen.h
+++ vcglib/vcg/math/eigen.h
@@ -53,10 +53,10 @@ struct ei_to_vcgtype;
 }
 
 #include "base.h"
-#include "../../eigenlib/Eigen/LU"
-#include "../../eigenlib/Eigen/Geometry"
-#include "../../eigenlib/Eigen/Array"
-#include "../../eigenlib/Eigen/Core"
+#include <Eigen/LU>
+#include <Eigen/Geometry>
+#include <Eigen/Array>
+#include <Eigen/Core>
 
 // add support for unsigned char and short int
 namespace Eigen {
Index: vcglib/vcg/math/matrix44.h
===================================================================
--- vcglib.orig/vcg/math/matrix44.h
+++ vcglib/vcg/math/matrix44.h
@@ -30,8 +30,8 @@
 #include <vcg/space/point4.h>
 #include <vector>
 #include <iostream>
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/LU>
+#include <Eigen/Core>
+#include <Eigen/LU>
 
 namespace vcg {
 
Index: vcglib/vcg/math/polar_decomposition.h
===================================================================
--- vcglib.orig/vcg/math/polar_decomposition.h
+++ vcglib/vcg/math/polar_decomposition.h
@@ -26,8 +26,8 @@
 
 #include <vcg/math/matrix33.h>
 #include <vcg/math/matrix44.h>
-#include <eigenlib/Eigen/Dense>
-#include <eigenlib/Eigen/SVD>
+#include <Eigen/Dense>
+#include <Eigen/SVD>
 
 namespace vcg{
 
Index: vcglib/vcg/space/fitting3.h
===================================================================
--- vcglib.orig/vcg/space/fitting3.h
+++ vcglib/vcg/space/fitting3.h
@@ -29,8 +29,8 @@
 #include <vcg/math/matrix44.h>
 #include <vcg/math/matrix33.h>
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <Eigen/Core>
+#include <Eigen/Eigenvalues>
 
 namespace vcg {
 
Index: vcglib/vcg/space/point_matching.h
===================================================================
--- vcglib.orig/vcg/space/point_matching.h
+++ vcglib/vcg/space/point_matching.h
@@ -27,8 +27,8 @@
 #include <vcg/math/quaternion.h>
 #include <vcg/math/matrix44.h>
 
-#include <eigenlib/Eigen/Dense>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <Eigen/Dense>
+#include <Eigen/Eigenvalues>
 #include  <iostream>
 
 namespace vcg
Index: vcglib/vcg/math/quadric.h
===================================================================
--- vcglib.orig/vcg/math/quadric.h
+++ vcglib/vcg/math/quadric.h
@@ -26,7 +26,7 @@
 #include <vcg/space/point3.h>
 #include <vcg/space/plane3.h>
 #include <vcg/math/matrix33.h>
-#include <eigenlib/Eigen/Core>
+#include <Eigen/Core>
 
 namespace vcg {
 namespace math {
Index: vcglib/vcg/complex/algorithms/harmonic.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/harmonic.h
+++ vcglib/vcg/complex/algorithms/harmonic.h
@@ -24,7 +24,7 @@
 #define __VCGLIB_HARMONIC_FIELD
 
 #include <vcg/complex/complex.h>
-#include <eigenlib/Eigen/Sparse>
+#include Eigen/Sparse>
 
 namespace vcg {
 namespace tri {
Index: vcglib/vcg/complex/algorithms/implicit_smooth.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/implicit_smooth.h
+++ vcglib/vcg/complex/algorithms/implicit_smooth.h
@@ -23,7 +23,7 @@
 #ifndef __VCG_IMPLICIT_SMOOTHER
 #define __VCG_IMPLICIT_SMOOTHER
 
-#include <eigenlib/Eigen/Sparse>
+#include Eigen/Sparse>
 #include <vcg/complex/algorithms/mesh_to_matrix.h>
 #include <vcg/complex/algorithms/update/quality.h>
 #include <vcg/complex/algorithms/smooth.h>
