Index: vcglib/vcg/complex/algorithms/clustering.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/clustering.h	2014-12-17 10:48:11.672086701 +0100
+++ vcglib/vcg/complex/algorithms/clustering.h	2014-12-17 10:48:11.668086681 +0100
@@ -35,7 +35,7 @@
 #include <limits>
 
 // some stuff for portable hashes...
-#ifdef WIN32
+/*#ifdef WIN32
  #ifndef __MINGW32__
   #include <hash_map>
   #include <hash_set>
@@ -50,7 +50,9 @@
  #include <ext/hash_set>
  #define STDEXT __gnu_cxx
 #endif
-
+*/
+//include fix to make code portable
+#include <hashfix.h>
 
 
 namespace vcg{
@@ -74,10 +76,16 @@
 
 // needed for gcc compilation
 #ifndef _MSC_VER
-}} namespace STDEXT {
+}} namespace std {
+#if defined(HAVE_TR1) && !defined(HAVE_CXX11)
+  namespace tr1 {
+ #endif
   template <> struct hash<vcg::tri::HashedPoint3i>{
   inline	size_t	operator ()(const vcg::tri::HashedPoint3i &p) const {return size_t(p);}
 };
+#if defined(HAVE_TR1) && !defined(HAVE_CXX11)
+  }
+ #endif
 } namespace vcg{ namespace tri{
 #endif
 
Index: vcglib/vcg/space/index/spatial_hashing.h
===================================================================
--- vcglib.orig/vcg/space/index/spatial_hashing.h	2014-12-17 10:48:11.672086701 +0100
+++ vcglib/vcg/space/index/spatial_hashing.h	2014-12-17 10:48:11.668086681 +0100
@@ -29,7 +29,7 @@
 //#include <map>
 #include <vector>
 #include <algorithm>
-#ifdef _WIN32
+/*#ifdef _WIN32
  #ifndef __MINGW32__
   #include <hash_map>
   #define STDEXT stdext
@@ -48,7 +48,9 @@
 #define __DEPRECATED
 #endif
 #endif
-
+*/
+//include fix to make code portable
+#include <hashfix.h>
 
 namespace vcg{
 
