Index: vcglib/vcg/complex/algorithms/clustering.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/clustering.h	2015-10-08 16:56:34.525695718 +0200
+++ vcglib/vcg/complex/algorithms/clustering.h	2015-10-16 14:08:02.372987126 +0200
@@ -35,7 +35,7 @@
 #include <limits>
 
 // some stuff for portable hashes...
-#ifdef WIN32
+/*#ifdef WIN32
  #ifndef __MINGW32__
   #include <hash_map>
   #include <hash_set>
@@ -50,7 +50,9 @@
  #include <ext/hash_set>
  #define STDEXT __gnu_cxx
 #endif
-
+*/
+//include fix to make code portable
+#include <hashfix.h>
 
 
 namespace vcg{
Index: vcglib/vcg/space/index/spatial_hashing.h
===================================================================
--- vcglib.orig/vcg/space/index/spatial_hashing.h	2015-10-08 16:56:34.525695718 +0200
+++ vcglib/vcg/space/index/spatial_hashing.h	2015-10-08 16:56:34.521695698 +0200
@@ -29,7 +29,7 @@
 //#include <map>
 #include <vector>
 #include <algorithm>
-#ifdef _WIN32
+/*#ifdef _WIN32
  #ifndef __MINGW32__
   #include <hash_map>
   #define STDEXT stdext
@@ -48,7 +48,9 @@
 #define __DEPRECATED
 #endif
 #endif
-
+*/
+//include fix to make code portable
+#include <hashfix.h>
 
 namespace vcg{
 
