Index: vcglib/vcg/space/index/grid_util.h
===================================================================
--- vcglib.orig/vcg/space/index/grid_util.h
+++ vcglib/vcg/space/index/grid_util.h
@@ -27,12 +27,15 @@
 #include<vcg/space/index/base.h>
 #include<vcg/space/box3.h>
 #include <vcg/space/index/space_iterators.h>
+#include <cstdint>
+//#ifndef WIN32
 
 
-#ifndef WIN32
-#define __int64 long long
-#define __cdecl
-#endif
+#define __int64 int64_t
+//#define __cdecl 
+
+//#define __cdecl
+
 
 namespace vcg {
 
