Index: vcglib/vcg/complex/algorithms/create/emc_lookup_table.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/create/emc_lookup_table.h	2014-05-20 20:23:34.585134175 +0200
+++ vcglib/vcg/complex/algorithms/create/emc_lookup_table.h	2014-05-20 20:40:47.666256942 +0200
@@ -923,6 +923,6 @@
 			//=============================================================================
 
 		}; //end of class EMCLookUpTable
-	}; // end of namespace tri
-}; // end of namespace vcg
+	} // end of namespace tri
+} // end of namespace vcg
 #endif // __VCG_EMC_LOOK_UP_TABLE
Index: vcglib/vcg/complex/algorithms/create/extended_marching_cubes.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/create/extended_marching_cubes.h	2014-05-20 20:29:43.038961233 +0200
+++ vcglib/vcg/complex/algorithms/create/extended_marching_cubes.h	2014-05-20 20:40:47.666256942 +0200
@@ -248,7 +248,7 @@
             TRIMESH_TYPE	*_mesh;
             /*!
             */
-            bool	_initialized;;
+            bool	_initialized;
             /*!
             */
             bool	_finalized;
@@ -464,6 +464,6 @@
         // end of Doxygen documentation
 
     } // end of namespace tri
-}; // end of namespace vcg
+} // end of namespace vcg
 
 #endif // __VCG_EXTENDED_MARCHING_CUBES
Index: vcglib/vcg/complex/algorithms/create/marching_cubes.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/create/marching_cubes.h	2014-05-20 20:23:34.585134175 +0200
+++ vcglib/vcg/complex/algorithms/create/marching_cubes.h	2014-05-20 20:40:47.666256942 +0200
@@ -721,7 +721,7 @@
         /*! @} */
         //end of Doxygen documentation
 
-    }; // end of namespace tri
-}; // end of namespace vcg
+    } // end of namespace tri
+} // end of namespace vcg
 
 #endif //__VCG_MARCHING_CUBES
Index: vcglib/vcg/complex/algorithms/create/mc_lookup_table.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/create/mc_lookup_table.h	2014-05-20 20:23:34.585134175 +0200
+++ vcglib/vcg/complex/algorithms/create/mc_lookup_table.h	2014-05-20 20:40:47.662256920 +0200
@@ -2552,8 +2552,8 @@
 			//_____________________________________________________________________________
 
 		}; // end of class EMCLookUpTable
-	}; // end of namespace tri
-}; //end of namespace vcg
+	} // end of namespace tri
+} //end of namespace vcg
 
 
 
Index: vcglib/vcg/complex/algorithms/create/mc_trivial_walker.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/create/mc_trivial_walker.h	2014-05-20 20:23:34.585134175 +0200
+++ vcglib/vcg/complex/algorithms/create/mc_trivial_walker.h	2014-05-20 20:40:47.666256942 +0200
@@ -199,7 +199,7 @@
 		extractor.Finalize();
 		_volume = NULL;
 		_mesh		= NULL;
-	};
+	}
 
 	float V(int pi, int pj, int pk)
 	{
Index: vcglib/vcg/complex/used_types.h
===================================================================
--- vcglib.orig/vcg/complex/used_types.h	2014-05-20 20:23:34.557134031 +0200
+++ vcglib/vcg/complex/used_types.h	2014-05-20 20:40:47.658256906 +0200
@@ -90,6 +90,6 @@
 struct _Face  : public  Face<_UsedTypes>{};
 struct _HEdge : public  HEdge<_UsedTypes>{};
 
-};
+}
 
 #endif // USED_TYPES_H
Index: vcglib/vcg/space/intersection3.h
===================================================================
--- vcglib.orig/vcg/space/intersection3.h	2014-05-20 20:23:34.573134114 +0200
+++ vcglib/vcg/space/intersection3.h	2014-05-20 20:40:47.658256906 +0200
@@ -121,7 +121,7 @@
 			solution_count++;
 		}
 		return solution_count;
-	}; // end of IntersectionSegmentSphere
+	} // end of IntersectionSegmentSphere
 
 	
 	/*!
@@ -215,7 +215,7 @@
 		penetration_detected = (witness.SquaredNorm() <= (radius*radius));
 		witness += center;
 		return penetration_detected;
-	}; //end of IntersectionSphereTriangle
+	} //end of IntersectionSphereTriangle
 
   /// intersection between line and plane
   template<class T>
Index: vcglib/wrap/ply/plylib.cpp
===================================================================
--- vcglib.orig/wrap/ply/plylib.cpp	2014-05-20 20:23:34.581134150 +0200
+++ vcglib/wrap/ply/plylib.cpp	2014-05-20 20:40:47.662256920 +0200
@@ -3673,6 +3673,6 @@
 		output[io++]=a[ia++]; 
 	};
 	output[io]=0;
-};
+}
 } 
 }
