Index: vcglib/vcg/math/base.h
===================================================================
--- vcglib.orig/vcg/math/base.h	2014-12-17 11:14:58.812056078 +0100
+++ vcglib/vcg/math/base.h	2014-12-17 11:14:58.808056058 +0100
@@ -92,6 +92,7 @@
 #include <limits>
 #include <algorithm>
 
+using namespace std;//as suggested by Brian Ripley
 
   #ifdef __BORLANDC__
     float sqrtf (float v) {return sqrt(v);}
