Index: vcglib/vcg/math/base.h
===================================================================
--- vcglib.orig/vcg/math/base.h	2014-06-06 09:08:03.705623017 +0200
+++ vcglib/vcg/math/base.h	2014-06-06 09:08:03.701622997 +0200
@@ -92,6 +92,7 @@
 #include <limits>
 #include <algorithm>
 
+using namespace std;//as suggested by Brian Ripley
 
   #ifdef __BORLANDC__
     float sqrtf (float v) {return sqrt(v);}
