Index: vcglib/vcg/complex/exception.h
===================================================================
--- vcglib.orig/vcg/complex/exception.h	2015-10-19 11:04:33.411700027 +0200
+++ vcglib/vcg/complex/exception.h	2015-10-19 11:04:33.407700007 +0200
@@ -30,7 +30,7 @@
 public:
   MissingComponentException(const std::string &err):std::runtime_error(err)
   {
-    std::cout << "Missing Component Exception -" << err << "- \n";
+    //std::cout << "Missing Component Exception -" << err << "- \n";
   }
     virtual const char *what() const throw ()
     {
@@ -44,7 +44,7 @@
 public:
   MissingCompactnessException(const std::string &err):std::runtime_error(err)
   {
-    std::cout << "Lack of Compactness Exception -" << err << "- \n";
+    //std::cout << "Lack of Compactness Exception -" << err << "- \n";
   }
     virtual const char *what() const throw ()
     {
