Index: vcglib/vcg/space/index/kdtree/priorityqueue.h
===================================================================
--- vcglib.orig/vcg/space/index/kdtree/priorityqueue.h	2015-04-14 11:44:27.161733645 +0200
+++ vcglib/vcg/space/index/kdtree/priorityqueue.h	2015-04-14 11:44:27.157733625 +0200
@@ -105,7 +105,7 @@
 			{
 				if (weight<mElements[0].weight)
 				{
-					register int j, k;
+					int j, k;
 					j = 1;
 					k = 2;
 					while (k <= mMaxSize)
