Index: vcglib/vcg/complex/algorithms/create/mc_trivial_walker.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/create/mc_trivial_walker.h	2014-06-06 09:05:40.868914728 +0200
+++ vcglib/vcg/complex/algorithms/create/mc_trivial_walker.h	2014-06-06 10:37:56.156362736 +0200
@@ -90,7 +90,7 @@
     else v->P().Y() = (float) p1.Y();
     if(AxisVal==ZAxis) v->P().Z() = (float) p1.Z()*(1-u) + u*p2.Z();
     else v->P().Z() = (float) p1.Z();
-    this->IPfToPf(v->P(),v->P());
+    //this->IPfToPf(v->P(),v->P());
     if(VoxelType::HasNormal()) v->N() = V(p1).N()*(1-u) + V(p2).N()*u;
   }
 
