Index: vcglib/vcg/space/index/space_iterators.h
===================================================================
--- vcglib.orig/vcg/space/index/space_iterators.h	2015-06-23 10:56:31.898100146 +0200
+++ vcglib/vcg/space/index/space_iterators.h	2015-06-23 10:56:31.894100126 +0200
@@ -314,7 +314,7 @@
 		RayType r;							//ray to find intersections
 		Spatial_Idexing &Si;	  //reference to spatial index algorithm
 		bool end;								//true if the scan is terminated
-		INTFUNCTOR &int_funct;
+		INTFUNCTOR int_funct;
 		TMARKER tm;
 
 		std::vector<Entry_Type> Elems;					//element loaded from curren cell
