Index: vcglib/vcg/complex/algorithms/point_sampling.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/point_sampling.h	2015-04-14 11:44:27.789736759 +0200
+++ vcglib/vcg/complex/algorithms/point_sampling.h	2015-04-14 11:44:27.785736740 +0200
@@ -47,6 +47,7 @@
 #include <vcg/complex/algorithms/update/bounding.h>
 #include <vcg/complex/algorithms/update/flag.h>
 #include <vcg/space/segment2.h>
+#include <time.h>
 
 namespace vcg
 {
