Index: vcglib/vcg/complex/algorithms/point_sampling.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/point_sampling.h	2015-06-23 10:56:32.086101078 +0200
+++ vcglib/vcg/complex/algorithms/point_sampling.h	2015-06-23 10:56:32.082101058 +0200
@@ -47,6 +47,7 @@
 #include <vcg/complex/algorithms/update/bounding.h>
 #include <vcg/complex/algorithms/update/flag.h>
 #include <vcg/space/segment2.h>
+#include <time.h>
 
 namespace vcg
 {
