Index: vcglib/vcg/complex/algorithms/clean.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/clean.h	2015-10-19 11:05:30.651983866 +0200
+++ vcglib/vcg/complex/algorithms/clean.h	2015-10-19 11:05:30.647983847 +0200
@@ -452,7 +452,7 @@
     return count_removed;
   }
 
-  static int SplitSelectedVertexOnEdgeMesh(MeshType& m)
+  static void SplitSelectedVertexOnEdgeMesh(MeshType& m)
   {
     tri::RequireCompactness(m);
     tri::UpdateFlags<MeshType>::VertexClearV(m);
