Index: vcglib/vcg/complex/algorithms/clean.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/clean.h	2015-04-14 11:44:27.989737751 +0200
+++ vcglib/vcg/complex/algorithms/clean.h	2015-04-14 11:44:27.985737731 +0200
@@ -452,7 +452,7 @@
     return count_removed;
   }
 
-  static int SplitSelectedVertexOnEdgeMesh(MeshType& m)
+  static void SplitSelectedVertexOnEdgeMesh(MeshType& m)
   {
     tri::RequireCompactness(m);
     tri::UpdateFlags<MeshType>::VertexClearV(m);
