Index: vcglib/vcg/complex/algorithms/clean.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/clean.h
+++ vcglib/vcg/complex/algorithms/clean.h
@@ -453,7 +453,7 @@ public:
     return count_removed;
   }
 
-  static int SplitSelectedVertexOnEdgeMesh(MeshType& m)
+  static void SplitSelectedVertexOnEdgeMesh(MeshType& m)
   {
     tri::RequireCompactness(m);
     tri::UpdateFlags<MeshType>::VertexClearV(m);
