Index: vcglib/vcg/space/index/kdtree/kdtree.h
===================================================================
--- vcglib.orig/vcg/space/index/kdtree/kdtree.h	2015-10-19 11:05:30.839984799 +0200
+++ vcglib/vcg/space/index/kdtree/kdtree.h	2015-10-19 11:05:30.835984779 +0200
@@ -31,7 +31,11 @@
 #include <vector>
 #include <limits>
 #include <iostream>
+#ifdef HAVE_CXX11
 #include <cstdint>
+#else
+#include <stdint.h>
+#endif
 
 namespace vcg {
 
