Index: vcglib/vcg/math/base.h
===================================================================
--- vcglib.orig/vcg/math/base.h	2015-12-16 08:59:35.266304264 +0100
+++ vcglib/vcg/math/base.h	2015-12-16 08:59:55.386404034 +0100
@@ -190,7 +190,7 @@
 #if defined(_MSC_VER) // Microsoft Visual C++
 template<class T> int IsNAN(T t) {    return _isnan(t) || (!_finite(t)); }
 #elif defined(__GNUC__) // GCC
-template<class T> int IsNAN(T t) {    return isnan(t) || isinf(t); }
+  template<class T> int IsNAN(T t) {    return std::isnan(t) || std::isinf(t); }
 #else // generic
 
 template<class T> int IsNAN(T t)
