Index: vcglib/vcg/math/base.h
===================================================================
--- vcglib.orig/vcg/math/base.h
+++ vcglib/vcg/math/base.h
@@ -85,9 +85,14 @@ Edited Comments and GPL license
 
 #ifndef __VCGLIB_MATH_BASE
 #define __VCGLIB_MATH_BASE
-
+#include <config.h>
 #include <float.h>
+#ifdef HAVE_CXX11
+#include <cmath>
+using namespace std;
+#else
 #include <math.h>
+#endif
 #include <assert.h>
 #include <limits>
 #include <algorithm>
@@ -191,6 +196,7 @@ inline double  ToRad(const double &a){re
 template<class T> int IsNAN(T t) {    return _isnan(t) || (!_finite(t)); }
 #elif defined(__GNUC__) // GCC
 template<class T> int IsNAN(T t) {    return isnan(t) || isinf(t); }
+
 #else // generic
 
 template<class T> int IsNAN(T t)
