Index: vcglib/vcg/math/base.h
===================================================================
--- vcglib.orig/vcg/math/base.h
+++ vcglib/vcg/math/base.h
@@ -87,11 +87,11 @@ Edited Comments and GPL license
 #define __VCGLIB_MATH_BASE
 
 #include <float.h>
-#include <math.h>
+#include <cmath>
 #include <assert.h>
 #include <limits>
 #include <algorithm>
-
+using namespace std;
 
   #ifdef __BORLANDC__
     float sqrtf (float v) {return sqrt(v);}
