Index: vcglib/vcg/math/base.h
===================================================================
--- vcglib.orig/vcg/math/base.h
+++ vcglib/vcg/math/base.h
@@ -85,9 +85,10 @@ Edited Comments and GPL license
 
 #ifndef __VCGLIB_MATH_BASE
 #define __VCGLIB_MATH_BASE
-
+#include <config.h>
 #include <float.h>
 #include <math.h>
+#include <cmath.h>
 #include <assert.h>
 #include <limits>
 #include <algorithm>
@@ -190,7 +191,11 @@ inline double  ToRad(const double &a){re
 #if defined(_MSC_VER) // Microsoft Visual C++
 template<class T> int IsNAN(T t) {    return _isnan(t) || (!_finite(t)); }
 #elif defined(__GNUC__) // GCC
-template<class T> int IsNAN(T t) {    return isnan(t) || isinf(t); }
+#ifdef HAVE_CXX11
+  template<class T> int IsNAN(T t) {    return std::isnan(t) || std::isinf(t); }
+#else
+  template<class T> int IsNAN(T t) {    return isnan(t) || isinf(t); }
+#endif
 #else // generic
 
 template<class T> int IsNAN(T t)
