Index: vcglib/vcg/complex/algorithms/refine_loop.h
===================================================================
--- vcglib.orig/vcg/complex/algorithms/refine_loop.h
+++ vcglib/vcg/complex/algorithms/refine_loop.h
@@ -100,7 +100,7 @@ struct RegularLoopWeight {
                 .69908
         };
 
-        return (k<=12)?(1.0-bkPolar[k-3])/k:LoopWeight<SCALAR_TYPE>().beta(k);
+        return (k<=12 && k>=3)?(1.0-bkPolar[k-3])/k:LoopWeight<SCALAR_TYPE>().beta(k);
     }
 
     inline SCALAR_TYPE incidentRegular(int k) {
@@ -120,7 +120,7 @@ struct RegularLoopWeight {
                 .44579
         };
 
-        return (k<=12)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().incidentIrregular(k);
+        return (k<=12 && k>=3)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().incidentIrregular(k);
     }
     inline SCALAR_TYPE opposite(int k) {
         static SCALAR_TYPE bkPolar[] = {
@@ -136,7 +136,7 @@ struct RegularLoopWeight {
                 .19828
         };
 
-        return (k<=12)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().opposite(k);
+        return (k<=12 && k>=3)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().opposite(k);
     }
 };
 
@@ -156,7 +156,7 @@ struct ContinuityLoopWeight {
                 .70014
         };
 
-        return (k<=12)?(1.0-bkPolar[k-3])/k:LoopWeight<SCALAR_TYPE>().beta(k);
+        return (k<=12 && k>=3)?(1.0-bkPolar[k-3])/k:LoopWeight<SCALAR_TYPE>().beta(k);
     }
 
     inline SCALAR_TYPE incidentRegular(int k) {
@@ -176,7 +176,7 @@ struct ContinuityLoopWeight {
                 .2452
         };
 
-        return (k<=12)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().incidentIrregular(k);
+        return (k<=12 && k>=3)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().incidentIrregular(k);
     }
     inline SCALAR_TYPE opposite(int k) {
         static SCALAR_TYPE bkPolar[] = {
@@ -192,7 +192,7 @@ struct ContinuityLoopWeight {
                 .29934
         };
 
-        return (k<=12)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().opposite(k);
+        return (k<=12 && k>=3)?bkPolar[k-3]:LoopWeight<SCALAR_TYPE>().opposite(k);
     }
 };
 
