Index: vcglib/wrap/io_trimesh/export_stl.h
===================================================================
--- vcglib.orig/wrap/io_trimesh/export_stl.h
+++ vcglib/wrap/io_trimesh/export_stl.h
@@ -92,7 +92,7 @@ static int Save(SaveMeshType &m, const c
     if(binary)
     {
         // Write Header
-        char header[128]="VCG                                                                                                  ";
+        char header[256]="VCG                                                                                                  ";
         if(objectname)	strncpy(header,objectname,80);
         if(magicsMode)
         {
