Index: vcg/complex/algorithms/clustering.h
===================================================================
--- vcg/complex/algorithms/clustering.h	(revision 5054)
+++ vcg/complex/algorithms/clustering.h	(working copy)
@@ -35,24 +35,10 @@
 #include <limits>
 
 // some stuff for portable hashes...
-#ifdef WIN32
- #ifndef __MINGW32__
-  #include <hash_map>
-  #include <hash_set>
-  #define STDEXT stdext
- #else
-  #include <ext/hash_map>
-  #include <ext/hash_set>
-  #define STDEXT __gnu_cxx
- #endif
-#else
- #include <ext/hash_map>
- #include <ext/hash_set>
- #define STDEXT __gnu_cxx
-#endif
+#include <unordered_map>
+#include <unordered_set>
 
 
-
 namespace vcg{
 namespace tri{
 #define HASH_P0 73856093
@@ -74,10 +60,11 @@
 
 // needed for gcc compilation
 #ifndef _MSC_VER
-}} namespace STDEXT {
+}} 
+namespace std {
   template <> struct hash<vcg::tri::HashedPoint3i>{
-  inline	size_t	operator ()(const vcg::tri::HashedPoint3i &p) const {return size_t(p);}
-};
+    inline	size_t	operator ()(const vcg::tri::HashedPoint3i &p) const {return size_t(p);}
+  };
 } namespace vcg{ namespace tri{
 #endif
 
@@ -263,18 +250,13 @@
 
   BasicGrid<ScalarType> Grid;
 
-#ifdef _MSC_VER
-  STDEXT::hash_set<SimpleTri> TriSet;
-  typedef typename STDEXT::hash_set<SimpleTri>::iterator TriHashSetIterator;
-#else
   struct SimpleTriHashFunc{
     inline	size_t	operator ()(const SimpleTri &p) const {return size_t(p);}
   };
-  STDEXT::hash_set<SimpleTri,SimpleTriHashFunc> TriSet;
-  typedef typename STDEXT::hash_set<SimpleTri,SimpleTriHashFunc>::iterator TriHashSetIterator;
-#endif
+  std::unordered_set<SimpleTri,SimpleTriHashFunc> TriSet;
+  typedef typename std::unordered_set<SimpleTri,SimpleTriHashFunc>::iterator TriHashSetIterator;
 
-  STDEXT::hash_map<HashedPoint3i,CellType> GridCell;
+  std::unordered_map<HashedPoint3i,CellType> GridCell;
 
 
 	void AddPointSet(MeshType &m, bool UseOnlySelected=false)
@@ -317,7 +299,7 @@
 
   void SelectPointSet(MeshType &m)
   {
-        typename STDEXT::hash_map<HashedPoint3i,CellType>::iterator gi;
+    typename std::unordered_map<HashedPoint3i,CellType>::iterator gi;
                 UpdateSelection<MeshType>::VertexClear(m);
         for(gi=GridCell.begin();gi!=GridCell.end();++gi)
     {
@@ -333,7 +315,7 @@
         if (GridCell.empty()) return;
 
     Allocator<MeshType>::AddVertices(m,GridCell.size());
-    typename STDEXT::hash_map<HashedPoint3i,CellType>::iterator gi;
+    typename std::unordered_map<HashedPoint3i,CellType>::iterator gi;
     int i=0;
     for(gi=GridCell.begin();gi!=GridCell.end();++gi)
     {
@@ -353,7 +335,7 @@
     if (GridCell.empty())  return;
 
     Allocator<MeshType>::AddVertices(m,GridCell.size());
-    typename STDEXT::hash_map<HashedPoint3i,CellType>::iterator gi;
+    typename std::unordered_map<HashedPoint3i,CellType>::iterator gi;
     int i=0;
     for(gi=GridCell.begin();gi!=GridCell.end();++gi)
     {
Index: vcg/space/index/kdtree/kdtree.h
===================================================================
--- vcg/space/index/kdtree/kdtree.h	(revision 5054)
+++ vcg/space/index/kdtree/kdtree.h	(working copy)
@@ -56,14 +56,14 @@
 	{
 		union {
                         //standard node
-			struct {
+			struct ano{
 				Scalar splitValue;
 				unsigned int firstChildId:24;
 				unsigned int dim:2;
 				unsigned int leaf:1;
 			};
                         //leaf
-			struct {
+			struct ano0{
 				unsigned int start;
 				unsigned short size;
 			};
Index: vcg/space/index/spatial_hashing.h
===================================================================
--- vcg/space/index/spatial_hashing.h	(revision 5054)
+++ vcg/space/index/spatial_hashing.h	(working copy)
@@ -29,27 +29,8 @@
 //#include <map>
 #include <vector>
 #include <algorithm>
-#ifdef _WIN32
- #ifndef __MINGW32__
-  #include <hash_map>
-  #define STDEXT stdext
- #else
-  #include <ext/hash_map>
-  #define STDEXT __gnu_cxx
- #endif
-#else  // We are in the *nix gcc branch
-#if (__GNUC__ ==4) && (__GNUC_MINOR__ > 3) && (defined(__DEPRECATED))
-  #undef __DEPRECATED // since gcc 4.4 <ext/hash_map> was deprecated and generate warnings. Relax Deprecation Just for this...
-  #define ___WE_UNDEFINED_DEPRECATED__
-#endif
- #include <ext/hash_map>
- #define STDEXT __gnu_cxx
-#if defined(___WE_UNDEFINED_DEPRECATED__)
-#define __DEPRECATED
-#endif
-#endif
+#include <unordered_map>
 
-
 namespace vcg{
 
 
@@ -101,7 +82,7 @@
 	// the hash index directly the grid structure.
 	// We use a MultiMap because we need to store many object (faces) inside each cell of the grid.
 
-	typedef typename STDEXT::hash_multimap<Point3i, ObjType *, HashFunctor> HashType;
+	typedef typename std::unordered_multimap<Point3i, ObjType *, HashFunctor> HashType;
 	typedef typename HashType::iterator HashIterator;
 	HashType hash_table; // The real HASH TABLE **************************************
 
