Index: vcg/space/index/spatial_hashing.h
===================================================================
--- vcg/space/index/spatial_hashing.h	(revision 5055)
+++ vcg/space/index/spatial_hashing.h	(working copy)
@@ -29,7 +29,7 @@
 //#include <map>
 #include <vector>
 #include <algorithm>
-#ifdef _WIN32
+/*#ifdef _WIN32
  #ifndef __MINGW32__
   #include <hash_map>
   #define STDEXT stdext
@@ -48,8 +48,9 @@
 #define __DEPRECATED
 #endif
 #endif
+*/
+#include <hashfix.h>
 
-
 namespace vcg{
 
 
Index: vcg/complex/algorithms/clustering.h
===================================================================
--- vcg/complex/algorithms/clustering.h	(revision 5055)
+++ vcg/complex/algorithms/clustering.h	(working copy)
@@ -35,7 +35,7 @@
 #include <limits>
 
 // some stuff for portable hashes...
-#ifdef WIN32
+/*#ifdef WIN32
  #ifndef __MINGW32__
   #include <hash_map>
   #include <hash_set>
@@ -50,9 +50,10 @@
  #include <ext/hash_set>
  #define STDEXT __gnu_cxx
 #endif
+*/
+#include <hashfix.h>
 
 
-
 namespace vcg{
 namespace tri{
 #define HASH_P0 73856093
@@ -74,10 +75,16 @@
 
 // needed for gcc compilation
 #ifndef _MSC_VER
-}} namespace STDEXT {
+}} namespace std {
+ #ifdef HAVE_TR1
+  namespace tr1 {
+ #endif
   template <> struct hash<vcg::tri::HashedPoint3i>{
   inline	size_t	operator ()(const vcg::tri::HashedPoint3i &p) const {return size_t(p);}
 };
+ #ifdef HAVE_TR1
+  }
+ #endif
 } namespace vcg{ namespace tri{
 #endif
 
