Index: vcg/space/index/kdtree/kdtree.h
===================================================================
--- vcg/space/index/kdtree/kdtree.h	(revision 5055)
+++ vcg/space/index/kdtree/kdtree.h	(working copy)
@@ -54,20 +54,20 @@
 
 	struct Node
 	{
-		union {
+	  //union {
                         //standard node
-			struct {
+			//struct {
 				Scalar splitValue;
 				unsigned int firstChildId:24;
 				unsigned int dim:2;
 				unsigned int leaf:1;
-			};
+		  //};
                         //leaf
-			struct {
+		  //struct {
 				unsigned int start;
 				unsigned short size;
-			};
-                };
+			  //	};
+	  // };
 	};
 	typedef std::vector<Node> NodeList;
 
