Index: vcg/space/index/grid_util.h
===================================================================
--- vcg/space/index/grid_util.h	(revision 5055)
+++ vcg/space/index/grid_util.h	(working copy)
@@ -68,11 +68,17 @@
 #include<vcg/space/index/base.h>
 #include<vcg/space/box3.h>
 #include <vcg/space/index/space_iterators.h>
+#include <config.h>
 
-
 #ifndef WIN32
-#define __int64 long long
-#define __cdecl 
+ #ifdef HAVE_INTTYPES
+  #include <inttypes.h>
+  #define __int64 int64_t
+  //#define __cdecl 
+ #else
+  #define __int64 long long
+  //#define __cdecl 
+ #endif
 #endif
 
 namespace vcg {
