Index: vcg/space/index/grid_util.h
===================================================================
--- vcg/space/index/grid_util.h	(revision 5204)
+++ vcg/space/index/grid_util.h	(working copy)
@@ -29,10 +29,18 @@
 #include <vcg/space/index/space_iterators.h>
 
 
-#ifndef WIN32
-#define __int64 long long
-#define __cdecl
-#endif
+#include <config.h>
+ 
+//#ifndef WIN32
+#ifdef HAVE_INTTYPES
+  #include <inttypes.h>
+  #define __int64 int64_t
+  //#define __cdecl 
+ #else
+  #define __int64 long long
+  //#define __cdecl 
+ #endif
+//#endif
 
 namespace vcg {
 
